src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let newSeparator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    } else {
+      newSeparator = separator;
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === null)
+      || (this[i] === undefined
+        && (i !== this.length - 1))) {
+        result += newSeparator;
+      } else if (this[i] !== undefined && i === this.length - 1) {
+        result += this[i];
+      } else if (i !== this.length - 1) {
+        result += this[i];
+        result += newSeparator;
+      }
```

Comment:
you can use `if` without `if else`

```suggestion
      if (this[i] !== undefined && this[i] !== null ) {
        result += this[i];
      }
      
      if (i !== this.length - 1) {
        result += newSeparator;
      }
```